packages/@aws-cdk/aws-ec2/lib/vpc-peering-connection.ts

```diff
@@ -0,0 +1,132 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { CfnVPCPeeringConnection } from './ec2.generated';
+import { IVpc } from './vpc';
+
+/**
+ * VPC with which you are creating the VPC peering connection.
+ */
+export class PeerVpc {
+  constructor(
+    /**
+     * The ID of the VPC with which you are creating the VPC peering connection
+     */
+    public readonly vpcId: string,
+
+    /**
+     * The Region code for the accepter VPC
+     *
+     * @default - The Region in which you make the request
+     */
+    public readonly region?: string,
+
+    /**
+     * The AWS account ID of the owner of the accepter VPC
+     *
+     * @default - Your AWS account ID
+     */
+    public readonly account?: string,
+  ) {}
+}
+
+/**
+ * Options for constructing VpcPeeringConnection
+ */
+export interface VpcPeeringConnectionOptions {
+  /**
+   * The name of VPC peering connections
+   *
+   * @default - generated by CloudFormation
+   */
+  readonly vpcPeeringConnectionName?: string
+
+  /**
+   * VPC with which you are creating the VPC peering connection.
+   */
+  readonly peerVpc: PeerVpc
+
+  /**
+   * The role, which is required when you are peering a VPC in a different AWS account.
+   *
+   * @default - conditional
+   */
+  readonly role?: iam.IRole
+}
+
+/**
+ * Props for constructing VpcPeeringConnection
+ */
+export interface VpcPeeringConnectionProps extends VpcPeeringConnectionOptions {
```

Comment:
You probably only want to make a distinction between `VpcPeeringConnectionProps` and `VpcPeeringConnectionOptions` if you're going to add a `vpc.addPeeringConnection()` method to `Vpc` itself (or maybe even to `IVpc`?)

Reply:
I intented that.